### PR TITLE
feat(cli): New cli to bundle the assets into cotar file.

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/cli/src/cli/config/action.bundle.assets.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const outputTar = output.split('.co')[0];
+
+    const tarBuilder = new TarBuilder(outputTar);
+
+    const files = await fsa.toArray(fsa.list(input));
+    // Ensure files are put into the same order
+    files.sort((a, b) => a.localeCompare(b));
+    logger.info({ output: outputTar, files: files.length }, 'Tar:Create');
+
+    for (const file of files) await tarBuilder.write(file, await fsa.read(file));
```
Comment (blacha) [inline packages/cli/src/cli/config/action.bundle.assets.ts:62]: this file name might need to be shortened, what does the tar archive look like?

Comment (Wentao-Kuang) [reply packages/cli/src/cli/config/action.bundle.assets.ts:62]: Yes, do we need to be short as `assets/fonts/...` or just `fonts/...`?

Comment (Wentao-Kuang) [reply packages/cli/src/cli/config/action.bundle.assets.ts]: Updated. By doing this. We also can move the fonts.json to root directory in cotar file.

Comment (blacha) [reply packages/cli/src/cli/config/action.bundle.assets.ts]: I think just leave the json file as is, and we can just update the server to point at the right spot?

File: packages/cli/package.json [hunk 2]
```diff
@@ -45,6 +45,8 @@
     "@basemaps/sprites": "^6.29.0",
     "@chunkd/fs": "^8.4.0",
     "@cogeotiff/core": "^7.0.0",
+    "@cotar/core": "^5.4.0",
```
Comment (blacha) [inline packages/cli/package.json:48]: can you check to see if we are using another version of cotar somewhere else? the yarn lock changes seem excessive!

File: packages/cli/src/cli/config/action.bundle.assets.ts (hunk 1 above)
Comment (Wentao-Kuang) [reply packages/cli/src/cli/config/action.bundle.assets.ts]: It was in `fonts/fonts.json` after build. Then we move it to `fonts.json` in the cotar. Shall we still keep it in the `fonts/font.json`?

File: packages/cli/src/cli/config/action.bundle.assets.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    const outputTar = output.split('.co')[0];
+
+    const tarBuilder = new TarBuilder(outputTar);
+
+    const files = await fsa.toArray(fsa.list(input));
+    // Ensure files are put into the same order
+    files.sort((a, b) => a.localeCompare(b));
+    logger.info({ output: outputTar, files: files.length }, 'Tar:Create');
+
+    for (const file of files) {
```
Comment (blacha) [inline packages/cli/src/cli/config/action.bundle.assets.ts:62]: I think the best option would be to use something like
```typescript
const basePath = path.resolve(input)
filePath = file.replace(basePath, '')
```

File: packages/cli/src/cli/config/action.bundle.assets.ts (hunk 1 above)
Comment (blacha) [reply packages/cli/src/cli/config/action.bundle.assets.ts]: yeah this bundling logic should replicate the excat file system we should change it in the filesystem if we want it changed!

File: packages/cli/src/cli/config/action.bundle.assets.ts (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/cli/src/cli/config/action.bundle.assets.ts:63]: Fixed. Thanks you.

File: packages/cli/package.json (hunk 2 above)
Comment (Wentao-Kuang) [reply packages/cli/package.json:48]: Fixed.

File: packages/cli/src/cli/config/action.bundle.assets.ts (hunk 3 above)
Comment (blacha) [reply packages/cli/src/cli/config/action.bundle.assets.ts:63]: I didnt actually check this, does this actually give us the correct tar output?

**Review by blacha — APPROVED** [review]

File: packages/cli/src/cli/config/action.bundle.assets.ts (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/cli/src/cli/config/action.bundle.assets.ts:64]: Leading / removed now. Thanks.

**Pull request merged**